Refactor/upgrade bootstrap 4.1

mpourismaiel opened the pull request:
**What this PR does / why we need it**:
Upgrades bootstrap version and adds webpack to automate scss/js build processes

**Which issue this PR fixes**:
fixes #68 

**Special notes for your reviewer**:
- **Don't merge this when the label contains wip.**
- Webpack still does not support js files.
- Styles need checking out. I'm not sure everything works.

**Release note**:
<!--
Optional one line note for this specific change, that can be used in a release-note or changelog.
-->
```release-note

```

stp-ip commented:
Also needs a fix for the conflicting file.

mpourismaiel commented:
Does everything works as before? I tested it as far as I could and everything looked okay. If it seems okay to you too I'm going to add js to the pipeline as well.

mpourismaiel commented on package.json:
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "syna",
+  "version": "0.8.0",
+  "description": "Open Source Theme for Hugo Static Site Generator based on Bootstrap 4",
```
Fixed in 8117f87e0f195558574ebc1a7d430aaa39b0c9d4

mpourismaiel commented on static-main/styles/index.scss:
```diff
@@ -0,0 +1,4 @@
+@import '~bootstrap/scss/bootstrap.scss';
+@import './bootstrap-overwrite/overwrite-color.scss';
+@import './form.scss';
```
If it's okay I would do that in another PR. This PR is way too huge unfortunately.

mpourismaiel commented:
@stp-ip I added js files to Webpack. Tested them and everything seemed in order. If you can check everything again yourself that would be awesome. I consider this PR done if everything works.

stp-ip commented:
On a first rough review I think it's good. Will do a full one tomorrow.
There is a comment for clarification and one thing that seems to be broken is the "scroll to top button" icon.

mpourismaiel commented:
Fixed fontawesome bug. Also jquery form validator's npm package is idiotic. I ignored the error it raises in webpack concerning "no static export". 

Also merged master to my branch and it didn't raise any conflicts. I have removed `static/css/syna.css` so either there were no changed since I branched off or I have no idea why Git is being stupid.
